dashboard/app/models/ability.rb

```diff
@@ -61,7 +61,7 @@ def initialize(user)
       can :create, Follower, student_user_id: user.id
       can :destroy, Follower, student_user_id: user.id
 
-      if user.hint_access? || user.teacher?
+      if user.permissions.where(permission: UserPermission::HINT_ACCESS).any? || user.teacher?
```

Comment:
Done. Nice, good to know those methods exist. Before sending out the migration for review, I'll update that PR to take advantage of `User.permission=` and `User.delete_permission`.